typings/pinejs-client-core.d.ts

```diff
@@ -46,7 +46,14 @@ export type InferAssociatedResourceType<T> =
 export type SelectableProps<T> =
 	// This allows us to get proper results when T is any/AnyObject, otherwise this returned never
 	PropsOfType<T, ReverseNavigationResource<object>> extends StringKeyof<T>
-		? StringKeyof<T>
+		? StringKeyof<T> extends PropsOfType<T, ReverseNavigationResource<object>>
```

Comment:
This is necessary to keep things working nicely now that `PropsOfType` no longer incorrectly returns `'a' | 'b' | undefined` when there are optional properties, meaning that `PropsOfType<T, ReverseNavigationResource<object>> extends StringKeyof<T>` now matches (previously it failed due to the `undefined`)